db/prefixdb_iterator.go

```diff
@@ -0,0 +1,132 @@
+package db
```

Comment:
should this be in a different package, like the mem one?

Reply:
It could be, but hasn't caused any import issues yet.

Reply:
I think it would be better to keep it consistent, either as a package or both under db/. Thoughts?

Reply:
I'm would rather leave them inside `db`, with the other utility stuff.

Reply:
My first impression was that this should be a separate package too...

Reply:
Agree - we are breaking the API so it's good time to have a bit better code organization.

Reply:
Okay will move it